### PR TITLE
ZF3 support

### PR DESCRIPTION
**Pull request by webimpress**

### CODE REVIEW

File: src/ContentValidationListener.php [hunk 1]
```diff
@@ -49,7 +49,7 @@ class ContentValidationListener implements ListenerAggregateInterface, EventMana
     protected $inputFilters = [];
 
     /**
-     * @var CallbackHandler[]
+     * @var callable[]
```
Comment (weierophinney) [inline src/ContentValidationListener.php]: Not true in v2 usage, though, which is why I recommend using the `ListenerAggregateTrait`, and removing the property altogether.

File: src/ContentValidationListener.php [hunk 2]
```diff
@@ -237,12 +238,15 @@ public function onRoute(MvcEvent $e)
 
         $e->setParam('ZF\ContentValidation\InputFilter', $inputFilter);
 
+        $event = clone $e;
```
Comment (weierophinney) [inline src/ContentValidationListener.php]: We don't actually need to clone it. In v2, if you pass an event in, it sets the name directly on the instance. As such, you can just call `$e->setName()`.

File: src/ContentValidationListenerFactory.php [hunk 3]
```diff
@@ -1,27 +1,26 @@
...
  */
 
 namespace ZF\ContentValidation;
 
-use Zend\ServiceManager\FactoryInterface;
-use Zend\ServiceManager\ServiceLocatorInterface;
+use Interop\Container\ContainerInterface;
 
-class ContentValidationListenerFactory implements FactoryInterface
+class ContentValidationListenerFactory
```
Comment (weierophinney) [inline src/ContentValidationListenerFactory.php]: This is a BC break for anybody who might have been extending the factory previously; this is why I've been updating the factories to implement both `__invoke()` and `createService()` signatures. (Doing so is *required* if you want to accept options during invocation for v2 as well.)

File: test/InputFilter/InputFilterPluginTest.php [hunk 4]
```diff
@@ -1,18 +1,23 @@
...
 {
-    public function setUp()
+    protected function setUp()
     {
+        parent::setUp();
+
         $this->event = $event = new MvcEvent();
 
-        $controller = $this->getMock('Zend\Mvc\Controller\AbstractController');
+        $controller = $this->createMock('Zend\Mvc\Controller\AbstractController');
```
Comment (weierophinney) [inline test/InputFilter/InputFilterPluginTest.php]: `createMock()` doesn't work with PHPUnit v4.

**Comment by weierophinney** [comment]
I'm going to merge this along with #67.

File: test/InputFilter/InputFilterPluginTest.php (hunk 4 above)
Comment (webimpress) [inline test/InputFilter/InputFilterPluginTest.php]: yes, I know, I've bumped version PHPUnit to 5.4 - see `composer.json`

**Comment by weierophinney** [comment]
@webimpress I've merged locally already, and am resolving the few conflicts between the two versions.

**Comment by webimpress** [comment]
@weierophinney I've just updated my PR according to yours comments, thanks

**Comment by weierophinney** [comment]
Merged!

**Pull request closed without merging**